### DISCUSSION TITLE
generate epub from reader mode

### DISCUSSION THREAD

**Discussion by Monirzadeh** [go-shiori/shiori#605] [Feature requests]
An option to generate an EPUB version of links that shiori can generate reader mode version?
It helps to read long article in an e-book reader etc...

**Comment by blogbourri** [comment]
Agree!

**Comment by Monirzadeh** [comment]
@fmartingr hi 
I write an incomplete code to add this feature to shiori.
For now, I generate a EPUB version from bookmarks in database with all images etc and save that on disk.
But for saving part that I should choose between this 3 way.
1. EPUB each time generate with user request (not offline archive but simple. Always Need internet to download image with each EPUB generate request)
2. EPUB generate and save on disk once, then somehow user can request file download link
3. save byte of EPUB in database and can user request that.

As maintainer, which of them you prefer. Personally, I don't like method 1.

**Reply by fmartingr** [reply]
Hey @Monirzadeh ! Thats good to hear :) 

I'd go with option 2. Generate the ePub on request but cache it from disk. If we could generate it from the archived version even better. You can use the SHIORI_DIR content path to store them inside a folder, as we do with content. 

My only request would be decoupling the code as much as possible, since Im working on an API refactor that will separate logic into different domains, so that will make my work easier (I found myself without a personal computer to use thats why I have been quiet for so long, but I plan to solve that in the coming weeks). 

You can open the PR with your work and we can continue the discussion there if you want. 

Thank you!

**Reply by Monirzadeh** [reply]
Hi.
I will wrap up that and send a PR (keep it as Draft until all parts complete) soon.
I save that on disk (as you mention) but need more tuning, some part missing like cover and language tag, etc.
For now, I try to understand API.

**Comment by fmartingr** [comment]
Initial support has been added and it's currently in the latest release candidate [v1.5.5-rc.1](https://github.com/go-shiori/shiori/releases/tag/v1.5.5-rc.1). Huge kudos to @Monirzadeh for working on this.

**Reply by Monirzadeh** [reply]
Thank you.
If you have any specific idea for this section, I am being happy to hear that.
